rs-react-app/src/pages/Home/components/Details/Details.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import style from './Details.module.css';
+import { useParams } from 'react-router';
+import { useFetchStarWarsPerson } from '../SearchInput/hooks/useFetchStarWarsPerson';
+import { Loader } from '../../../../sharedComponents/Loader/Loader';
+
+export const Details: FC = () => {
+  const navigate = useNavigate();
+
+  const { itemName: characterName, pageNumber } = useParams<{
+    itemName?: string;
+    pageNumber?: string;
+  }>();
+
+  const { item, isLoading, isError, setFetchedPersonToState } =
+    useFetchStarWarsPerson(characterName || '');
+
+  useEffect(() => {
+    if (!characterName) return;
+
+    setFetchedPersonToState();
+  }, [characterName]);
+
+  const handleClose = () => {
+    navigate(`/page/${pageNumber || 1}`, { replace: true });
+  };
+
+  return (
+    <div className={style.detailsContainer} onClick={handleClose}>
+      <div className={style.detailsPanel} onClick={(e) => e.stopPropagation()}>
+        {isError && <h3>Something went wrong. Try again later</h3>}
```

Comment:
What's purpose of this `handleClose`..? It's quite overridden by next container by `stopPropagation`